### ISSUE TITLE
Realm crash when using GroupActivities

### ISSUE THREAD

**Issue by mantovanirian14** [realm/realm-swift#8046]
### How frequently does the bug occur?

All the time

### Description

We have an app that implements SharePlay, realm happens to be crashed when we have the shareplay code, but when we completely remove the implementation, the code is not happening anymore

### Stacktrace & log output
<img width="1252" alt="Screenshot 2022-11-29 at 16 40 34" src="https://user-images.githubusercontent.com/119405173/204494025-9e500edd-4553-410d-98b3-bf26d10cc528.png">

### Can you reproduce the bug?

Yes, always

### Version

10.14

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

< iOS 15

### Build environment

Xcode version: 14.1
Dependency manager and version:  Swift Package Manager

**Comment by leemaguire** [comment]
Hi @mantovanirian14 can you send us a small app which reproduces the issue?

**Comment by mantovanirian14** [comment]
sure this is the link to the app https://drive.google.com/file/d/1multxjGR72T2LcJbXONZapP-XPX73VEk/view?usp=sharing I also found out when @Published is removed the crash didn't happen, hope it helps

**Comment by leemaguire** [comment]
@mantovanirian14 I don't see the GroupActivities capability in this sample app, could you provide a sample app which can hit the crash straight out of the box? Or provide more details on how to hit the crash?

**Comment by mantovanirian14** [comment]
@leemaguire have you run the project in iOS 13/14? turns out when using @Published with GroupActivity make realm crash, but when you remove the @Published the crash didn't happen

**Comment by leemaguire** [comment]
@mantovanirian14 share play is only available for iOS 15+. We need detailed steps to invoke the crash on our end.

**Comment by mantovanirian14** [comment]
yes share play is only available for iOS 15, the bug didn't need to implement shareplay fully,but happens when we have this code @Published with GroupActivities
<img width="744" alt="Screenshot 2022-11-30 at 20 09 30" src="https://user-images.githubusercontent.com/119405173/204804678-a0b7461d-b268-4c41-ac88-e4d912f85f9e.png">

to make it clearer:
Running the app with @Published -> Crash
https://user-images.githubusercontent.com/119405173/204804981-44822fd6-e0c9-4ff7-b6d9-3bc2fbe7f515.mov

Running the app without @Published -> No Crash
https://user-images.githubusercontent.com/119405173/204805193-db959b1c-8cbe-44c9-9c19-23bdd261706e.mov

I hope the video helps

**Comment by dianaafanador3** [comment]
We use `[objc_copyclasslist](https://developer.apple.com/documentation/objectivec/1418762-objc_copyclasslist)`, in our schema discovery implementation, this function allow us to get a list of pointers to all registered class in the target which will be used to identify the ones that inherit from `Object` to create the database schema.
There seems to be crash when we use  
 (`objc_copyclasslist is part of the `[Objective-C Runtime](https://developer.apple.com/documentation/objectivec/objective-c_runtime)`), and a property like the one in your sample code `@Published var groupSession: GroupSession<Activity>?`, which seems to be fixed on iOS >= 15.0.
You can test this adding the following lines to your sample, and remove any `RealmSwift` reference in your code and you will still get the crash
```
var count = UInt32(0)
if let classList = objc_copyClassList(&count) {
}
```
Unfortunately this an apple issue and we cannot do something about it in our side.

**Comment by dianaafanador3** [comment]
As a suggestion I would use a custom schema setting your object types `objectTypes` in your realm configuration, this would avoid using the function mentioned in the comment above.
